src/processing_provider/QgepSwmm.py

```diff
@@ -0,0 +1,504 @@
+
+    def create_swmm_views(self):
+        """
+        Create QGEP-SWMM views
+        """
+
+        con = psycopg2.connect(service=self.service)
+        cur = con.cursor()
+        cur.execute(open(os.path.join(self.db_model_path, "vw_swmm_conduits.sql"), "r").read())
+        cur.execute(open(os.path.join(self.db_model_path, "vw_swmm_junctions.sql"), "r").read())
+        cur.execute(open(os.path.join(self.db_model_path, "vw_swmm_outfalls.sql"), "r").read())
+        cur.execute(open(os.path.join(self.db_model_path, "vw_swmm_pumps.sql"), "r").read())
+        cur.execute(open(os.path.join(self.db_model_path, "vw_swmm_storages.sql"), "r").read())
+        cur.execute(open(os.path.join(self.db_model_path, "vw_swmm_subcatchments.sql"), "r").read())
+        cur.execute(open(os.path.join(self.db_model_path, "vw_swmm_xsections.sql"), "r").read())
+        cur.execute(open(os.path.join(self.db_model_path, "vw_swmm_losses.sql"), "r").read())
+        cur.execute(open(os.path.join(self.db_model_path, "vw_swmm_coordinates.sql"), "r").read())
+        cur.execute(open(os.path.join(self.db_model_path, "vw_swmm_vertices.sql"), "r").read())
+        cur.execute(open(os.path.join(self.db_model_path, "vw_swmm_polygons.sql"), "r").read())
+        cur.execute(open(os.path.join(self.db_model_path, "vw_swmm_tags.sql"), "r").read())
+        cur.execute(open(os.path.join(self.db_model_path, "vw_swmm_aquifers.sql"), "r").read())
+        cur.execute(open(os.path.join(self.db_model_path, "vw_swmm_landuses.sql"), "r").read())
+        con.commit()
+
+        return
+
+    def delete_swmm_tables(self):
+        """
+        Delete swmm tables
+        """
+        con = psycopg2.connect(service=self.service)
+        cur = con.cursor()
+        cur.execute("DROP TABLE IF EXISTS qgep_swmm.conduits")
+        cur.execute("DROP TABLE IF EXISTS qgep_swmm.junctions")
+        cur.execute("DROP TABLE IF EXISTS qgep_swmm.outfalls")
+        cur.execute("DROP TABLE IF EXISTS qgep_swmm.pumps")
+        cur.execute("DROP TABLE IF EXISTS qgep_swmm.storages")
+        cur.execute("DROP TABLE IF EXISTS qgep_swmm.subcatchments")
+        cur.execute("DROP TABLE IF EXISTS qgep_swmm.subareas")
+        cur.execute("DROP TABLE IF EXISTS qgep_swmm.infiltration")
+        cur.execute("DROP TABLE IF EXISTS qgep_swmm.xsections")
+        cur.execute("DROP TABLE IF EXISTS qgep_swmm.losses")
+        cur.execute("DROP TABLE IF EXISTS qgep_swmm.coordinates")
+        cur.execute("DROP TABLE IF EXISTS qgep_swmm.vertices")
+        cur.execute("DROP TABLE IF EXISTS qgep_swmm.polygons")
+        cur.execute("DROP TABLE IF EXISTS qgep_swmm.raingages")
+        cur.execute("DROP TABLE IF EXISTS qgep_swmm.tags")
+        cur.execute("DROP TABLE IF EXISTS qgep_swmm.aquifers")
+        cur.execute("DROP TABLE IF EXISTS qgep_swmm.landuses")
+        cur.execute("DROP TABLE IF EXISTS qgep_swmm.coverages")
+        con.commit()
+        return
+
+    def create_fill_swmm_tables(self):
```

Comment:
What's the reason for these tables? Should/could we work directly with the views?

Reply:
Yes we could. But my guess was that it could be useful to have a copy in order to have a snapshot of the network used for a study. 
But this snapshot is also in the swmm input file.
What do you think?

Reply:
The advantage of the input files is, that they can be copied and archived somewhere :+1: 
I would probably remove those tables for now. We can easily reintroduce them if we have a good workflow / use case where we really do need them for a strong need.

Reply:
A snapshot is important, as in the meantime changes could be made to the system, while the simulation process is going on. We need also the reference on which network state we have based our simulation.

Reply:
@sjib do you mean the files are not sufficient?

Reply:
We could suppress these tables for now and in the future we could add a functionality to generate de snapshot in a more standard format (gpkg)?

Reply:
I would like to have some feedback from current and future users on this @urskaufmann @GruBoe , Koni, Triform what do you think? 
@m-kuhn: Do the tables do any harm right now ?

Reply:
@sjib the "harm" they do is that they build an API without a clear requirement. Things that enter our codebase need maintenance, so everything comes at a cost. There are also different possibilities like "materialized views" and we didn't consider any pros and cons on which of these approaches to take.

I would drop it for now and see if we are all happy without it instead of having theoretical discussions without a complete understanding of the implications here.

Reply:
ok. good for me.

Reply:
As I understand: I have never made a copy of the database when I exported a UND/HGF-File for calculations in MOUSE (I think thats isthe question here). First idea: All the necessary data is in the exported file(s), so why make another copy of it?

Reply:
These tables are not created anymore. Let wait for the users feedback to see if the .inp snapshot is sufficient or if they need another way to generate a snapshot.

Reply:
I guess that for the moment it will be sufficient to have the imp-File.